src/test/java/teammates/sqlui/webapi/UpdateFeedbackSessionLogsActionTest.java

```diff
@@ -0,0 +1,207 @@
+        mockLogsProcessor.insertFeedbackSessionLog(student2, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusSeconds(900).toEpochMilli());
+
+        // Gap is larger than spam filter
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(), startTime.toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER + 1).toEpochMilli());
+
+        UpdateFeedbackSessionLogsAction action = getAction();
+        action.execute();
+
+        // method returns all logs regardless of params
+        List<FeedbackSessionLogEntry> expected =
+                mockLogsProcessor.getOrderedFeedbackSessionLogs("", "", 0, 0, "");
+
+        verify(mockLogic).createFeedbackSessionLogs(
+                argThat(filteredLogs -> isEqualExceptId(expected, filteredLogs)));
+    }
+
+    @Test
+    public void testExecute_recentLogsWithSpam_someLogsCreated()
+            throws EntityAlreadyExistsException, InvalidParametersException {
+        // Gap is smaller than spam filter
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(), startTime.toEpochMilli());
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER - 2).toEpochMilli());
+
+        // Filters multiple logs within one spam window
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER - 1).toEpochMilli());
+
+        // Correctly adds new log after filtering
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER + 1).toEpochMilli());
+
+        // Filters out spam in the new window
+        mockLogsProcessor.insertFeedbackSessionLog(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER + 2).toEpochMilli());
+
+        UpdateFeedbackSessionLogsAction action = getAction();
+        action.execute();
+
+        List<FeedbackSessionLogEntry> expected = new ArrayList<>();
+        expected.add(new FeedbackSessionLogEntry(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(), startTime.toEpochMilli()));
+        expected.add(new FeedbackSessionLogEntry(student1, feedbackSession1,
+                FeedbackSessionLogType.ACCESS.getLabel(),
+                startTime.plusMillis(SPAM_FILTER + 1).toEpochMilli()));
```

Comment:
Very small nit, will it be possible to create the objects and use your comments as variable names, it'll make it slightly more readable. E.g `correctlyAdded..` or `filteredOut..`. Then you'll be able to add the objects into `expected` here too. ID will probably be then equal too.